File: app/src/test/java/seveida/firetvforreddit/SubredditRepoTest.java
package seveida.firetvforreddit;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import io.reactivex.functions.Consumer;
import retrofit2.Retrofit;
import seveida.firetvforreddit.domain.objects.SubredditDetails;
import seveida.firetvforreddit.domain.objects.ThreadMetadata;
import seveida.firetvforreddit.response.objects.SubredditResponse;

import static org.junit.Assert.*;

@RunWith(RobolectricTestRunner.class)
public class SubredditRepoTest {

    @Test
    public void t() {
        Retrofit retrofit = RetrofitProvider.create();

        SubredditRepo repo = new SubredditRepo(retrofit);

        repo.getSubreddit("boobs")
                .subscribe(subreddit -> {
                    subreddit.threadMetadataList.forEach(thread -> {
                        System.out.println(thread.title);
                    });
                });

        try {
            Thread.sleep(10_000);
        } catch (Exception e) {

        }

    }

}

File: app/src/test/java/seveida/firetvforreddit/SubredditRepoTest.kt
package seveida.firetvforreddit

import baron.severin.io.RetrofitProvider
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import baron.severin.io.SubredditRepo

@RunWith(RobolectricTestRunner::class)
class SubredditRepoTest {

    @Test
    fun t() {
        val retrofit = RetrofitProvider.create()

        val repo = SubredditRepo(retrofit)

        repo.getSubreddit("aww")
            .subscribe { (_, threadMetadataList) -> threadMetadataList.forEach { (_, _, title) -> println(title) } }

        try {
            Thread.sleep(10000)
        } catch (e: Exception) {

        }

    }

}